common/data_refinery_common/models/attributes.py

```diff
@@ -0,0 +1,90 @@
+from django.db import models
+
+from data_refinery_common.models.ontology_term import OntologyTerm
+
+BOOL = "bool"
+INT = "int"
+FLOAT = "float"
+ONTOLOGY_TERM = "ont"
+
+
+class AbstractAttribute(models.Model):
+    """This is an abstract class that defines all of the properties of a single
+    attribute on either a sample or an experiment. We then subclass this to
+    associate attributes with either an experiment or a sample."""
+
+    name = models.ForeignKey("OntologyTerm", on_delete=models.CASCADE, related_name="+")
+    unit = models.ForeignKey("OntologyTerm", on_delete=models.CASCADE, related_name="+", null=True)
+    probability = models.FloatField(null=True)
+    source = models.TextField()
+
+    value_type = models.TextField(
+        choices=[
+            (BOOL, "Boolean"),
+            (INT, "Integer"),
+            (FLOAT, "Float"),
+            (ONTOLOGY_TERM, "Ontology Term"),
+        ],
+    )
+    value = models.TextField()
+
+    class Meta:
+        abstract = True
+
+    def to_dict(self):
+        rendered = {
+            "name": self.name.to_dict(),
+            "unit": None if self.unit is None else self.unit.to_dict(),
+            "probability": "unknown" if self.probability is None else self.probability,
```

Comment:
Is this better than reporting none?

Reply:
That is a good question, I am not sure. I felt like reporting `None` might make it seem like the probability was 100% rather than unreported.